docs/guides/code_push_release.md

```diff
@@ -0,0 +1,252 @@
+---
+sidebar_position: 2
+title: 🚢 Release a Code Push App
+description: Release a Code Push App
+---
+
+# Releasing an code push app
+
+This guide walks you through releasing a code push app to the Play Store and applying a patch to that release.
+
+## Prerequisites
+
+This guide assumes that you have an existing Shorebird app. If you don't have one, you can create one by following the [code push quickstart](./code_push_quickstart/) guide.
+
+The app we will be releasing in this guide is [`Time Shift`](https://play.google.com/store/apps/details?id=dev.shorebird.u_shorebird_clock), our demo code push app. ([source](https://github.com/shorebirdtech/time_shift/))
+
+## Creating a Release
+
+### Determine next release version
+
+We will start by running `shorebird releases list` to see the current set of releases:
+
+```
+bryanoltman@boltman ~/Shorebird/time_shift (main)
+⑆ shorebird releases list
+🚀 Releases (51751336-6a7c-4972-b4ec-8fc1591fb2b3)
+┌─────────┬──────┐
+│ Version │ Name │
+├─────────┼──────┤
+│ 1.0.1   │ --   │
+├─────────┼──────┤
+│ 1.0.2+1 │ --   │
+├─────────┼──────┤
+│ 1.0.2+5 │ --   │
+└─────────┴──────┘
+```
+
+From this, we can see that the most recent release is `1.0.2+5`. This corresponds with what we see in the Play Store console:
+
+![ReleaseVersion](https://github.com/shorebirdtech/docs/assets/581764/e6b6c276-49de-4142-8f32-dbf5e41379fa)
+
+The release we are going to create is `1.0.3+6`.
+
+### Make code changes
+
+Version `1.0.3+6` of Time Shift will be a small patch that changes the default clock face from `particle` to `generative`.
+
+To make this change, we will edit `lib/main.dart`:
+
+```diff
+   final clock = ClockFace.values.firstWhere(
+     (clock) => clock.name == clockName,
+-    orElse: () => ClockFace.particle,
++    orElse: () => ClockFace.generative,
+   );
+```
+
+We run the app using `shorebird run` to make sure this change does what we expect.
+
+After verifying, we will commit this change and push it to GitHub:
+
+```sh
+git add lib/main.dart
+git commit -m "Change default clock face to generative"
+git push
+```
+
+[On GitHub](https://github.com/shorebirdtech/time_shift/commit/d1fe9451aa18a775163bce95dd9dab551aaf6259)
+
+### Update app version
+
+We will now update the app version in `pubspec.yaml`:
+
+```diff
+name: time_shift
+description: Demo app showing Shorebird updates.
+publish_to: "none"
+
+-version: 1.0.2+5
++version: 1.0.3+6
+
+environment:
+  sdk: ">=2.19.4 <3.0.0"
```

Comment:
side-note we should probably upgrade to Dart 3 😄

Reply:
Agreed. Filed https://github.com/shorebirdtech/time_shift/issues/9.